From 1cb60e1fac28c19d0246172ccfbcbdfb60ab947b Mon Sep 17 00:00:00 2001
From: Pavel Nedev <pnedev@mm-sol.com>
Date: Wed, 16 Feb 2011 12:38:49 +0200
Subject: OMAP4/VOUT: Silence compilator warnings

Fixes the following compilator warnings:

drivers/media/video/omap/omap_vout.c: In function 'omap_tiler_virt_to_phys':
drivers/media/video/omap/omap_vout.c:1507: warning: unused variable 'til_addr'
drivers/media/video/omap/omap_vout.c:1506: warning: unused variable 'r'
drivers/media/video/omap/omap_vout.c: In function 'omap_vout_buffer_prepare':
drivers/media/video/omap/omap_vout.c:1548: warning: unused variable 'pdata'
drivers/media/video/omap/omap_vout.c: In function 'omap_tiler_virt_to_phys':
drivers/media/video/omap/omap_vout.c:1523: warning: control reaches end of non-void function

Change-Id: Ida0421221b7c2d60390190ddc4c313277115bc53
Signed-off-by: Pavel Nedev <pnedev@mm-sol.com>
---
 drivers/media/video/omap/omap_vout.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

Index: kernel/drivers/media/video/omap/omap_vout.c
===================================================================
--- kernel.orig/drivers/media/video/omap/omap_vout.c	2011-04-08 12:32:57.248346401 +0900
+++ kernel/drivers/media/video/omap/omap_vout.c	2011-04-08 15:03:40.557111854 +0900
@@ -1505,8 +1505,6 @@
 	pgd_t *pgd = NULL;
 	pmd_t *pmd = NULL;
 	pte_t *ptep = NULL, pte = 0x0;
-	s32 r = -1;
-	u32 til_addr = 0x0;
 	u32 arg = (u32)ptr;
 
 	pgd = pgd_offset(current->mm, arg);
@@ -1522,6 +1520,8 @@
 			}
 		}
 	}
+
+	return 0;
 }
 
 /*
@@ -1546,7 +1546,7 @@
 	dma_addr_t dmabuf;
 #endif
 	struct omap_vout_device *vout = q->priv_data;
-#ifdef CONFIG_PM
+#if defined(CONFIG_PM) && !defined(CONFIG_ARCH_OMAP4)
 	struct vout_platform_data *pdata =
 			(((vout->vid_dev)->v4l2_dev).dev)->platform_data;
 #endif
